psi4/src/psi4/libmints/eribase.cc

```diff
@@ -2036,6 +2190,7 @@ void TwoElectronInt::init_shell_pairs34() {
     if (use_shell_pairs_ == true) {
         // This assumes init_shell_pairs12 was called and precomputed the values.
         pairs34_ = pairs12_;
+        screenpairs34_ = screenpairs12_;
```

Comment:
Is there a way to get around this full copy of `screenpairs12_` to `screenpairs34_`?  The old `pairs12_` and `pairs12_` were pointers and thus free.  The new way could be costly.

Reply:
They could both be defined as shared_ptrs that point to the same object in this case, right?

Reply:
Yes, defining them as shared pointers to some object that holds the list of shell pairs (similar to how the vector does now) should do it, i think.  Looking great so far!

Reply:
Thanks! I've changed `pairs12_` to be a pointer to a vector of vectors. The initialization is pretty ugly, but it works:
    ```pairs12_ = std::make_shared<std::vector<std::vector<ShellPair>>>(std::vector<std::vector<ShellPair>>(basis2()->nshell(), std::vector<ShellPair>(basis2()->nshell())));```
Elements of the object are accessed by doing:
```(*pairs12_)[i][j]```
Are you all ok with this? I could alternatively put the vector of vectors in another struct. This would allow me to use `->` instead of `*` (which seems more in line with Psi4's style). The initialization of `pairs12_` could also be made to look a little cleaner.
Let me know what you all think!